src/python/fuzzing/coverage_uploader.py

```diff
@@ -41,20 +41,17 @@ def upload_testcases_if_needed(fuzzer_name, testcase_list, testcase_directory):
   if not bucket_name:
     return
 
-  # Only consider test cases in the output directory. We might upload too much
-  # if we search the data directory as well, or have missing resources.
-  # TODO(mbarbella): Support resources in data bundles.
-  testcase_list = [
-      os.path.relpath(testcase, testcase_directory)
-      for testcase in testcase_list
-      if testcase.startswith(testcase_directory)
-  ]
-  if not testcase_list:
-    return
-
-  # Bail out if this batch of test cases is too large.
-  directory_size = shell.get_directory_size(testcase_directory)
-  if directory_size >= MAX_TESTCASE_DIRECTORY_SIZE:
+  files_list = []
+  has_testcases_in_testcase_directory = False
+  has_testcases_in_data_directory = False
+  for testcase_path in testcase_list:
```

Comment:
There is an issue with `TESTCASES_PER_DAY` limit enforcement. basically, we have some fuzzers that manage to upload many thousands of testcases even though the limit if `1000` -- I can show you the number on my desktop. I think we should either:

1) break out of the loop, if `len(files_list) >= TESTCASES_PER_DAY`
2) write only up to `TESTCASES_PER_DAY` elements from `files_list` after the loop

We can also consider shuffling either `testcase_list` or `files_list` (not sure if it's needed, maybe it's already random order?)

Otherwise, the corpus is skewed towards those fuzzers that generate a lot of inputs and upload 10-20K instead of 1K. If we account for the limit in this function, we'll still have a race, but should have much better numbers anyway.

Reply:
Fixed